File: java/socketd/src/main/java/org/noear/socketd/transport/core/Frames.java
package org.noear.socketd.transport.core;

import org.noear.socketd.SocketD;
import org.noear.socketd.transport.core.entity.EntityDefault;
import org.noear.socketd.transport.core.internal.MessageDefault;

/**
 * 帧工厂
 *
 * @author noear
 * @since 2.0
 * */
public class Frames {
    /**
     * 构建连接帧
     *
     * @param url 连接地址
     */
    public static final Frame connectFrame(String sid, String url) {
        EntityDefault entity = new EntityDefault();
        //添加框架版本号
        entity.putMeta(EntityMetas.META_SOCKETD_VERSION, SocketD.version());
        return new Frame(Flag.Connect, new MessageDefault().sid(sid).topic(url).entity(entity));
    }

    /**
     * 构建连接确认帧
     *
     * @param connectMessage 连接消息
     */
    public static final Frame connackFrame(Message connectMessage) {
        EntityDefault entity = new EntityDefault();
        //添加框架版本号
        entity.meta(EntityMetas.META_SOCKETD_VERSION, SocketD.version());
        return new Frame(Flag.Connack, new MessageDefault().sid(connectMessage.getSid()).topic(connectMessage.getTopic()).entity(entity));
    }

    /**
     * 构建 ping 帧
     */
    public static final Frame pingFrame() {
        return new Frame(Flag.Ping, null);
    }

    /**
     * 构建 pong 帧
     */
    public static final Frame pongFrame() {
        return new Frame(Flag.Pong, null);
    }

    /**
     * 构建关闭帧（一般用不到）
     */
    public static final Frame closeFrame() {
        return new Frame(Flag.Close, null);
    }
}

File: python/socket_websocket/socketd/core/Frames.py
from socketd.core.Costants import Flag
import socketd.core.SocketD as SocketD
from socketd.core.module.EntityDefault import EntityDefault
from socketd.core.module.Frame import Frame
from socketd.core.module.MessageDefault import MessageDefault


class Frames:

    @staticmethod
    def connectFrame(sid, url):
        entity = EntityDefault()
        entity.put_meta("META_SOCKETD_VERSION", SocketD.SocketD.version())
        return Frame(Flag.Connect, MessageDefault().set_sid(sid).set_topic(url).set_entity(entity))

    @staticmethod
    def connackFrame(connectMessage):
        entity = EntityDefault()
        entity.put_meta("META_SOCKETD_VERSION", SocketD.SocketD.version())
        return Frame(Flag.Connack, MessageDefault().set_sid(connectMessage.sid).set_topic(connectMessage.topic).set_entity(entity))

    @staticmethod
    def pingFrame():
        return Frame(Flag.Ping, None)

    @staticmethod
    def pongFrame():
        return Frame(Flag.Pong, None)

    @staticmethod
    def closeFrame():
        return Frame(Flag.Close, None)